contracts/CarryPublicTokenCrowdsale.sol

```diff
@@ -174,14 +187,23 @@ contract CarryPublicTokenCrowdsale is CappedCrowdsale, Pausable {
         );
 
         // See also the comment on the individualMaxCaps above.
-        uint256 latestTimestamp = 0;
         uint256 individualMaxWei = 0;
-        for (uint256 i = 0; i < individualMaxCaps.length; i++) {
+        for (uint i = 0; i < individualMaxCaps.length; i++) {
+            uint256 capTimestamp = individualMaxCaps[i].timestamp;
             // solium-disable-next-line security/no-block-members
-            if (individualMaxCaps[i].timestamp <= block.timestamp &&
-                individualMaxCaps[i].timestamp > latestTimestamp) {
-                latestTimestamp = individualMaxCaps[i].timestamp;
+            if (capTimestamp <= block.timestamp) {
                 individualMaxWei = individualMaxCaps[i].maxWei;
+            } else {
+                // Optimize gas consumption by trimming timestamps no more used.
```

Comment:
Does this mean that the first contributor(per stage) might need to increase gas limit?

Reply:
Probably yes, to a certain degree, but not much.  As the existing logic already had iterated over the far future caps, the total cap of the worst case is not that increased.